### PR TITLE
feat: add authentication and webpack local proxy

### PR DESCRIPTION
**Pull request by sahil143**
Jira ticket:  https://issues.redhat.com/browse/HAC-5812

### CODE REVIEW

**Comment by gbenhaim** [comment]
@yftacherzog @avi-biton after we get this one merged I'll update konflux-ci to use the new image.

**Comment by gbenhaim** [comment]
@sahil143 thanks for working on it.
Do we need any unittest for this change?

File: .env [hunk 1]
```diff
@@ -0,0 +1,4 @@
+AUTH_URL=https://127.0.0.1:9443/
+REGISTRATION_URL=https://127.0.0.1:9443/
+PROXY_URL=https://127.0.0.1:9443/
+PROXY_WEBSOCKET_URL=wss://127.0.0.1:9443/
```
Comment (yftacherzog) [inline .env:4]: Can we add a newline at end of file?

File: src/main.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#root {
+  height: 100%;
+}
```
Comment (yftacherzog) [inline src/main.scss:3]: Can we add a newline at end of file?

**Review by yftacherzog — COMMENTED** [review]
@sahil143 @gbenhaim should this PR be towards the main branch or the new-ui branch?

**Comment by gbenhaim** [comment]
@yftacherzog the new image will be integrated in the `new-ui` branch since it still doesn't have all the functionality.

**Comment by yftacherzog** [comment]
> @yftacherzog the new image will be integrated in the `new-ui` branch since it still doesn't have all the functionality.

@sahil143 can you edit the PR so it's directed to the new-ui branch instead of main?

**Comment by sahil143** [comment]
> > @yftacherzog the new image will be integrated in the `new-ui` branch since it still doesn't have all the functionality.
> 
> @sahil143 can you edit the PR so it's directed to the new-ui branch instead of main?

There isn't any `new-ui` branch in this repo.

**Comment by yftacherzog** [comment]
> > > @yftacherzog the new image will be integrated in the `new-ui` branch since it still doesn't have all the functionality.
> > 
> > 
> > @sahil143 can you edit the PR so it's directed to the new-ui branch instead of main?
> 
> There isn't any `new-ui` branch in this repo.

My apologies. I was sure I was looking at konflux-ci/konflux-ci  🤦

**Comment by sahil143** [comment]
> @sahil143 thanks for working on it. Do we need any unittest for this change?

Not right now. Some implementations will change. I will be adding unit tests in the follow-up PR.

**Pull request merged**